iohk/UpdateProposal.hs

```diff
@@ -454,45 +485,42 @@ updateProposalUploadS3 :: CommandOptions -> Shell ()
 updateProposalUploadS3 opts@CommandOptions{..} = do
   params@UpdateProposalConfig2{..} <- loadParams opts
   void $ doCheckConfig params
-  echo "*** Hashing installers with sha256sum"
-  sha256 <- getHashes sha256sum cfgInstallersResults
-  echo "*** Hashing installers with cardano-auxx"
-  hashes <- getHashes (cardanoHashInstaller opts) cfgInstallersResults
   printf ("*** Uploading installers to S3 bucket "%s%"\n") cmdUpdateBucket
-  urls <- uploadInstallers cmdUpdateBucket cfgInstallersResults hashes
+  urls <- uploadInstallers cmdUpdateBucket cfgInstallersResults cfgInstallerHashes
   printf ("*** Uploading signatures to same S3 bucket.\n")
   signatures <- uploadSignatures cmdUpdateBucket cfgInstallersResults
   printf ("*** Writing "%fp%"\n") (versionFile opts)
-  let dvis = makeDownloadVersionInfo cfgInstallersResults urls hashes sha256 signatures
+  let dvis = makeDownloadVersionInfo cfgInstallersResults urls cfgInstallerHashes cfgInstallerSHA256 signatures
   liftIO $ writeVersionJSON (versionFile opts) dvis
-  storeParams opts (UpdateProposalConfig3 params hashes)
+  storeParams opts (UpdateProposalConfig3 params)
 
-uploadInstallers :: Text -> InstallersResults -> InstallerHashes -> Shell (Text, Text)
-uploadInstallers bucket res InstallerHashes{..} = runAWS' $ do
-  darwin <- needResult Mac64 res $ upload cfgInstallerDarwin
-  windows <- needResult Win64 res $ upload cfgInstallerWindows
-  pure (darwin, windows)
+uploadInstallers :: Text -> InstallersResults -> InstallerHashes -> Shell (ArchMap Text)
+uploadInstallers bucket res hashes = runAWS' $ needResults res upload
   where
-    upload hash ci = do
+    upload arch ci = do
+      let hash = lookupArch arch hashes
       printf ("***   "%s%"  "%fp%"\n") hash (ciResultLocalPath ci)
       uploadHashedInstaller bucket (ciResultLocalPath ci) (globalResult res) hash
 
 -- | Apply a hashing command to all the installer files.
 getHashes :: (FilePath -> Shell Text) -> InstallersResults -> Shell InstallerHashes
 getHashes getHash res = do
-  cfgInstallerDarwin <- check Mac64 =<< needResult Mac64 res resultHash
-  cfgInstallerWindows <- check Win64 =<< needResult Win64 res resultHash
-  pure $ InstallerHashes{..}
+  linux64 <- check Win64 =<< needResult Linux64 res resultHash
```

Comment:
s/Win64/Linux64 on this line

Reply:
Ah yes thanks. I didn't have the energy to reformulate `needResult` but maybe I should to avoid this duplication.